### PR TITLE
Edb mnemonic monitor rebase

### PR DESCRIPTION
**Pull request by bhilbert4**
This PR is a test to get around problems with conflict resolution when rebasing the edb-mnemonic-monitor branch. I've added back the 3 MIRI data trending files that had recent updates, to hopefully remove them from the list of files with conflicts, in the hopes that the github conflict resolution button will come back. Eventually the data trending files need to be deleted, as they're not actually used.

### CODE REVIEW

**Comment by bhilbert4** [comment]
Hooray it worked! So much easier than working manually one commit at a time on the command line.

Items left to do:
- [x] Add EDB monitor pull down menu entry for niriss, nirspec, fgs (already have miri, nircam)
- [x] Re-remove the data_trending files
- [x] One final test on test server
- [x] Change list of instrument names to loop over back to JWST_INSTRUMENT_NAMES
- [x] set self._today to the correct value
- [x] set EDB_DEFAULT_PLOT_RANGE back to production value of 14
- [x] Add new database tables to instrument-based table lists in database_interface
- [x] Add lock
- [ ] Remove NIRISS entries that are producing too many points

**Comment by bhilbert4** [comment]
@mfixstsci I want to remove the NIRISS entries that are causing plots to have way too many points and slowing the browser, but in terms of the code, this is finally finally ready for review.

File: jwql/tests/test_edb.py [hunk 1]
```diff
@@ -17,27 +17,107 @@
 
         pytest -s test_edb.py
 """
-
+from datetime import datetime
```
Comment (mfixstsci) [inline jwql/tests/test_edb.py:20]: @bhilbert4 thank you for adding tests!

File: jwql/tests/test_edb.py [hunk 2]
```diff
@@ -61,3 +147,143 @@ def test_get_mnemonics():
 
     mnemonic_dict = get_mnemonics(mnemonics, start_time, end_time)
     assert len(mnemonic_dict) == len(mnemonics)
+
+
+def test_add():
```
Comment (mfixstsci) [inline jwql/tests/test_edb.py]: @bhilbert4 I am being picky here but if could move this test to the top of the module to be in alphabetical order, that would be great.

File: jwql/tests/test_edb_telemetry_monitor.py [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        output = etm_utils.get_averaging_time_duration(inval)
+        assert output == outval
+
+    bad_strings = ["7_years", "nonsense"]
+    for inval in bad_strings:
+        with pytest.raises(ValueError) as e_info:
+            output = etm_utils.get_averaging_time_duration(inval)
+
+
+def test_get_query_duration():
```
Comment (mfixstsci) [inline jwql/tests/test_edb_telemetry_monitor.py]: @bhilbert4 same here with alphabetizing

File: jwql/tests/test_edb_telemetry_monitor.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+                     datetime.timedelta(days=1), datetime.timedelta(days=1)]
+    for inval, outval in zip(in_strings, expected_vals):
+        output = etm_utils.get_query_duration(inval)
+        assert output == outval
+
+    with pytest.raises(ValueError) as e_info:
+        output = etm_utils.get_query_duration("bad_string")
+
+
+def test_key_check():
```
Comment (mfixstsci) [inline jwql/tests/test_edb_telemetry_monitor.py]: Same here with alphabetizing

File: jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js.map [hunk 5]
```diff
@@ -0,0 +1 @@
...\n        this.element.removeAttribute('aria-describedby')\n        $(this.element).trigger(this.constructor.Event.HIDDEN)\n        if (this._popper !== null) {\n          this._popper.destroy()\n        }\n\n        if (callback) {\n          callback()\n        }\n      }\n\n      $(this.element).trigger(hideEvent)\n\n      if (hideEvent.isDefaultPrevented()) {\n        return\n      }\n\n      $(tip).removeClass(ClassName.SHOW)\n\n      // If this is a touch-enabled device we remove the extra\n      // empty mouseover listeners we added for iOS support\n      if ('ontouchstart' in document.documentElement) {\n        $(document.body).children().off('mouseover', null, $.noop)\n      }\n\n      this._activeTrigger[Trigger.CLICK] = false\n      this._activeTrigger[Trigger.FOCUS] = false\n      this._activeTrigger[Trigger.HOVER] = false\n\n      if ($(this.tip).hasClass(ClassName.FADE)) {\n        const transitionDuration = Util.getTransitionDurationFromElement(tip)\n\n        $(tip)\n          .one(Util.TRANSITION_END, complete)\n          .emulateTransitionEnd(transitionDuration)\n      } else {\n        complete()\n      }\n\n      this._hoverState = ''\n    }\n\n    update() {\n
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js.map]: Is there a reason we need to include this script in the page as opposed to importing it into base?

File: jwql/website/apps/jwql/static/js/popper.min.js.map [hunk 6]
```diff
@@ -0,0 +1 @@
...,GAFjBiV,EAKVrR,EAAc,QAAN4E,KAAiB,KAAjBA,CAAyB,SACjC1E,EAAc,OAAN4E,KAAgB,MAAhBA,CAAyB,QAKjCyM,EAAmB7I,EAAyB,WAAzBA,OAYX,QAAV1I,IAG4B,MAA1BrC,KAAajB,SACT,CAACiB,EAAauD,YAAd,CAA6BT,EAAQZ,OAErC,CAACuR,EAAiBzQ,MAAlB,CAA2BF,EAAQZ,OAGrCY,EAAQb,MAEF,OAAVM,IAC4B,MAA1BvC,KAAajB,SACR,CAACiB,EAAasD,WAAd,CAA4BR,EAAQV,MAEpC,CAACqR,EAAiB1Q,KAAlB,CAA0BD,EAAQV,MAGpCU,EAAQX,KAEbqR,kDAEc,OACA,IACT1I,WAAa,gBACf,IAEC+I,GAAsB,QAAVxR,IAAqB,CAAC,CAAtBA,CAA0B,EACtCyR,EAAuB,OAAVvR,IAAoB,CAAC,CAArBA,CAAyB,OAC5BN,GAJX,MAKWE,GALX,GAME2I,WAAgBzI,MAAAA,MAInBiK,GAAa,eACFpD,EAAKtD,SADH,WAKd0G,mBAAiCpD,EAAKoD,cACtC7J,eAAyByG,EAAKzG,UAC9BsR,kBAAmB7K,EAAKpG,OAALoG,CAAa+I,MAAU/I,EAAK6K,eIsKtC,mBAAA,GAkBT,QAlBS,GAwBT,OAxBS,CAlPD,YA4RD,OAEH,GAFG,WAAA,IM9Sd,WAAyC,UAK7B7K,EAAK2H,QAAL3H,CAAczD,OAAQyD,EAAKzG,UAIvByG,EAAK2H,QAAL3H,CAAczD,OAAQyD,EAAKoD,YAGrCpD,EAAKoI,YAALpI,EAAqBjD,OAAOC,IAAPD,CAAYiD,EAAK6K,WAAjB9N,EAA8BW,UAC3CsC,EAAKoI,aAAcpI,EAAK6K,eNiSxB,QMjRd,mBAME,IAEMhM,GAAmBuB,QAA8CC,EAAQC,aAAtDF,EAKnB1D,EAAY6D,EAChBF,EAAQ3D,SADQ6D,OAKhBF,EAAQZ,SAARY,CAAkBG,IAAlBH,CAAuBjE,iBALPmE,CAMhBF,EAAQZ,SAARY,CAAkBG,IAAlBH,CAAuB7D,OANP+D,WASX8C,aAAa,qBAIF,CAAE1C,SAAUN,EAAQC,aAARD,CAAwB,OAAxBA,CAAkC,UAA9C,KNuPN,uBAAA,CA5RC,CDdA"}
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/static/js/popper.min.js.map]: Same import question here, have we tried using the bootstrap style import?

File: jwql/website/apps/jwql/templates/edb_monitor.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+
+{% block content %}
+
+    <main role="main" class="container">
+
+                <h1>{{ inst }} EDB Telemetry Monitor</h1>
+                <hr>
+
+                <div id='myplot' width=100%>
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/templates/edb_monitor.html]: Being picky here, but maybe we give the div a more descriptive name here.

File: jwql/website/apps/jwql/urls.py [hunk 8]
```diff
@@ -70,6 +67,7 @@
     re_path(r'^(?P<inst>({}))/bad_pixel_monitor/$'.format(instruments), monitor_views.bad_pixel_monitor, name='bad_pixel_monitor'),
     re_path(r'^(?P<inst>({}))/bias_monitor/$'.format(instruments), monitor_views.bias_monitor, name='bias_monitor'),
     re_path(r'^(?P<inst>({}))/readnoise_monitor/$'.format(instruments), monitor_views.readnoise_monitor, name='readnoise_monitor'),
+    re_path(r'^(?P<inst>({}))/edb_monitor/$'.format(instruments), monitor_views.edb_monitor, name='edb_monitor'),
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/urls.py:70]: Like this change here, much more pythonic 👍

File: jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 /*!
-  * Bootstrap v4.0.0 (https://getbootstrap.com)
+  * Bootstrap v4.1.3 (https://getbootstrap.com/)
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js:2]: I wonder if we can do the same bootstrap style import here as well.

File: jwql/website/apps/jwql/forms.py [hunk 10]
```diff
@@ -458,19 +458,17 @@ def clean_search(self):
...
-        range_day = 1.
-        default_start_time = now + TimeDelta(delta_day, format='jd')
-        default_end_time = now + TimeDelta(delta_day + range_day, format='jd')
+        default_start_time = now + TimeDelta(3600., format='sec')
+        default_end_time = now
     else:
         # example for testing
-        default_start_time = Time('2019-04-02 00:00:00.000', format='iso')
-        default_end_time = Time('2019-04-02 00:01:00.000', format='iso')
+        default_start_time = Time('2022-06-20 00:00:00.000', format='iso')
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/forms.py:470]: Thanks for updating this.

**Review by mfixstsci — APPROVED** [review]
A lot of hard work here, thank you @bhilbert4 this looks great. A couple of comments and questions and then we can get this in. Thank you!

File: jwql/tests/test_edb.py (hunk 1 above)
Comment (bhilbert4) [reply jwql/tests/test_edb.py:20]: 👍

File: jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js.map (hunk 5 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js.map]: Probably not. I added this in response to warnings in the Chrome developer tools about a missing file. But that was before I understood that the browser would go download it on its own.

File: jwql/website/apps/jwql/static/js/popper.min.js.map (hunk 6 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/static/js/popper.min.js.map]: I'm betting we can get rid of this. Would be worth a test to be sure.

File: jwql/website/apps/jwql/forms.py (hunk 10 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/forms.py:470]: All it takes is 30 times entering dates by hand during testing to motivate a change. 😆

File: jwql/tests/test_edb.py (hunk 2 above)
Comment (bhilbert4) [reply jwql/tests/test_edb.py]: Done

File: jwql/tests/test_edb_telemetry_monitor.py (hunk 3 above)
Comment (bhilbert4) [reply jwql/tests/test_edb_telemetry_monitor.py]: Done

File: jwql/website/apps/jwql/templates/edb_monitor.html (hunk 7 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/templates/edb_monitor.html]: Done

File: jwql/tests/test_edb_telemetry_monitor.py (hunk 4 above)
Comment (bhilbert4) [reply jwql/tests/test_edb_telemetry_monitor.py]: Done

File: jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js.map (hunk 5 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js.map]: Removed from repo, and everything is still working.

File: jwql/website/apps/jwql/static/js/popper.min.js.map (hunk 6 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/static/js/popper.min.js.map]: Removed from repo, and everything is still working.

File: jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js (hunk 9 above)
Comment (bhilbert4) [reply jwql/website/apps/jwql/static/js/bootstrap.bundle.min.js:2]: This file was present in the repo prior to this PR. Looks like I just downloaded a more recent version.

**Comment by mfixstsci** [comment]
@bhilbert4 is all the testing for this PR finished? I am going to take a look on test and then merge if so.

**Pull request merged**